sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/CustomHeadersPolicy.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+public class CustomHeadersPolicy implements HttpPipelinePolicy {
+    private final Supplier<HttpHeaders> customHeaderSupplier;
+
+    public CustomHeadersPolicy(Supplier<HttpHeaders> customHeaderSupplier) {
```

Comment:
@JonathanGiles could this constructor be moved into `HttpHeadersPolicy` to allow for a dynamic header policy and a static header policy? This way we could resolve this requirement without introducing additional policies.

Reply:
Or might be AddHeadersPolicy, but not in RequestIDPolicy

Reply:
I'm all for reducing the number of policies. Please do consider this.